discovery-service/src/main/resources/discovery-service-log-messages.yml

```diff
@@ -21,21 +21,40 @@ messages:
+      action: "Review the static API definition directories and their setup. The static definition directories is specified as a launch parameter to a discovery service jar. The property key is: `apiml.discovery.staticApiDefinitionsDirectories`"
 
     - key: apiml.discovery.errorParsingStaticDefinitionFile
       number: ZWEAD701
       type: ERROR
-      text: "Error loading static Api definition file '%s'"
+      text: "Error loading static API definition file '%s'"
+      reason: "A problem occurred while reading (IO operation) of a specific static API definition file."
+      action: "Ensure that the file data is not corrupted or incorrectly encoded."
 
     - key: apiml.discovery.errorParsingStaticDefinitionData
       number: ZWEAD702
       type: WARNING
-      text: "Unable to process static Api definition data: '%s'"
+      text: "Unable to process static API definition data: '%s'"
+      reason: "A problem occurred while parsing a static API definition file."
+      action: "Review the mentioned static API definition file for errors.\n
+      Refer to the specific log message to see what is the exact cause of the problem:\n\n
+      - ServiceId is not defined in the file '%s'. The instance will not be created. Make sure to specify the ServiceId.\n
+      - The `instanceBaseUrls` parameter of %s is not defined. The instance will not be created. Make sure to specify the `InstanceBaseUrl` property.\n
+      - The API Catalog UI tile ID %s is invalid. The service %s will not have an API Catalog UI tile. Specify the correct catalog title ID.\n
+      - One of the instanceBaseUrl of %s is not defined. The instance will not be created. Make sure to specify the InstanceBaseUrl property.\n
+      - The URL %s does not contain a hostname. The instance of %s will not be created. The specified URL is malformed. Make sure to specify valid URL.\n
+      - The URL %s does not contain a port number. The instance of %s will not be created.\n
+      - The specified URL is missing a port number. Make sure to specify a valid URL.\n
+      - The URL %s is malformed. The instance of %s will not be created: The Specified URL is malformed. Make sure to specify a valid URL.\n
+      - The hostname of URL %s is unknown. The instance of %s will not be created: Thes specified hostname of the URL is invalid. Make sure to specify valid hostname.\n
+      - Invalid protocol. The specified protocol of the URL is invalid. Make sure to specify valid protocol."
 
     - key: apiml.discovery.errorReadingStaticDefinitionFolder
       number: ZWEAD703
       type: ERROR
-      text: "I/O problem occurred during reading static Api definition directory: '%s'"
+      text: "I/O problem occurred during reading static API definition directory: '%s'"
+      reason: "The property files of type .yml were not found."
+      action: "Review the paths for validity of your .yml files."
```

Comment:
Is the user who is reading the message the one who has created the `.yml` file. Will it be clear what *your .yml files` means?

Reply:
The message is changed now

Reply:
you were right, it was not clear